src/widgets/favoritewidget.cpp

```diff
@@ -50,12 +50,16 @@ QSize FavoriteWidget::sizeHint() const {
 }
 
 void FavoriteWidget::paintEvent(QPaintEvent* e) {
-  QStylePainter p(this);
+  // ignore text descent when vertically centering
+  QFontMetrics fontMetrics = QFontMetrics(this->font());
+  QRect rect = this->rect();
+  rect.setBottom(rect.bottom() - fontMetrics.descent());
```

Comment:
I'm afraid your suggested code does not work:
* `Qt::AlignVCenter` will center the pixmap in the QRect provided to `drawItemPixmap()`. That's why we need `this->rect()` here (the drawing area of the widget or its height specifically) and can't just use some arbitrary rectangle.
* Your code produces a rectangle with a height of 14px on my system (since the font descent is 2 px) and therefore would even cut the icon (since `QStyle.drawItemPixmap()` won't rescale the pixmap).

As long as we do not want to make guesses regarding the widget height (which we probably could, given that we make a `sizeHint()`) which would be pretty bad style and prone to errors in the future we somehow have to get a valid height...